src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -130,18 +166,19 @@ uORB::DeviceNode::copy(void *dst, unsigned &generation)
 			ATOMIC_ENTER;
 			const unsigned current_generation = _generation.load();
 
-			if (current_generation > generation + _queue_size) {
-				// Reader is too far behind: some messages are lost
-				generation = current_generation - _queue_size;
-			}
-
-			if ((current_generation == generation) && (generation > 0)) {
+			if (current_generation == generation) {
 				/* The subscriber already read the latest message, but nothing new was published yet.
 				* Return the previous message
 				*/
 				--generation;
 			}
 
+			// Compatible with normal and overflow conditions
+			if (!is_in_range(current_generation - _queue_size, generation, current_generation - 1)) {
+				// Reader is too far behind: some messages are lost
+				generation = current_generation - _queue_size;
+			}
+
 			memcpy(dst, _data + (_meta->o_size * (generation % _queue_size)), _meta->o_size);
 			ATOMIC_LEAVE;
 
```

Comment:
On the next line, the `if (generation < current_generation) {` condition needs to be removed, and is not necessary anymore.

Reply:
Forgot to modify it, now it is removed.